src/plugins/hyperbee/hyperbee.cpp

```diff
@@ -0,0 +1,1113 @@
+        size_t write_length = tmp_len_bytes;
+
+        if (write_length > VMI_PS_4KB - intra_page_offset)
+            write_length = VMI_PS_4KB - intra_page_offset;
+
+        if (access_ptrs[i])
+        {
+            fwrite((char*) access_ptrs[i] + intra_page_offset, write_length, 1, fp);
+            munmap(access_ptrs[i], VMI_PS_4KB);
+        }
+        else
+        {
+            // inaccessible page, pad with zeros to ensure proper alignment of the data
+            uint8_t zeros[VMI_PS_4KB] = {};
+            fwrite(zeros + intra_page_offset, write_length, 1, fp);
+        }
+
+        // this applies only to the first page
+        intra_page_offset = 0;
+        tmp_len_bytes -= write_length;
+    }
+
+    fclose(fp);
+
+    if (rename(plugin->tmp_file_path, file_path) != 0)
+    {
+        PRINT_DEBUG("[HYPERBEE] Failed to rename dump file\n");
+        goto error;
+    }
+
+    dump_success = true;
+    goto done;
+
+error:
+    PRINT_DEBUG("[HYPERBEE] Failed to dump memory\n");
+
+done:
+    g_free(access_ptrs);
+
+    return dump_success;
+}
+
+/**
+ * Saves the virtual address where the fault occurred.
+ */
+struct access_fault_result_t : public call_result_t
+{
+    access_fault_result_t() : call_result_t(), fault_va()
+    {}
+
+    addr_t fault_va;
+};
+
+/**
+ * If something was written at a monitored page this will swap the write trap to a exec trap.
+ * (right away after committing the page or again after the first time it was executed)
+ */
+static event_response_t write_faulted_cb(drakvuf_t drakvuf, drakvuf_trap_info_t* info)
+{
+
+    auto* ef_data = (struct fault_data_struct*) info->trap->data;
+
+    if (LOG_ALWAYS)
+    {
+        fmt::print(ef_data->plugin->m_output_format, "hyperbee", drakvuf, info,
+                   keyval("EventType", fmt::Qstr("writeframe")),
+                   keyval("FrameVA", fmt::Xval(ef_data->page_va)),
+                   keyval("TrapPA", fmt::Xval(info->trap_pa)),
+                   keyval("CR3", fmt::Xval(info->regs->cr3)),
+                   keyval("GFN", fmt::Xval(info->trap->memaccess.gfn))
+                  );
+    }
+
+    drakvuf_trap* exec_Trap = create_execute_trap(info->trap->memaccess.gfn, ef_data);
+
+    //Add and activate the trap
+    drakvuf_add_trap(drakvuf, exec_Trap);
+    //store the trap to be deleted in the end.
+    ef_data->plugin->traps.emplace(exec_Trap);
+
+    //Removes this trap and frees the memory
+    ef_data->plugin->traps.erase(info->trap);
+    drakvuf_remove_trap(drakvuf, info->trap, nullptr);
+
+    PRINT_DEBUG("[HYPERBEE] Replaced Write Trap X on GFN 0x%lx with Execute trap\n", info->trap->memaccess.gfn);
+
+    return VMI_EVENT_RESPONSE_NONE;
+}
+
+/**
+ * Creates a write trap, whose pointer is returned.
+ *
+ * @param info
+ * @param ef_data
+ */
+drakvuf_trap_t* create_write_trap(drakvuf_trap_info_t* info, fault_data_struct* ef_data)
+{
+
+    auto* write_trap = (drakvuf_trap_t*) g_malloc(sizeof(drakvuf_trap_t));
```

Comment:
You should use `g_malloc0` here, see remark about `create_execute_trap` for explanation.